photutils/detection/findstars.py

```diff
@@ -7,18 +7,20 @@
 import warnings
 import math
 import numpy as np
+import abc
 from astropy.table import Column, Table
 from astropy.utils.exceptions import AstropyUserWarning
 from astropy.stats import gaussian_fwhm_to_sigma
 from .core import _convolve_data, find_peaks
 
 
-__all__ = ['daofind', 'irafstarfind']
+class StarFinder(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def __call__(self, data):
```

Comment:
@mirca - No, definitely not.  The point of doing this is that `find_stars` is a clearer name, but `__call__` is more convenient.  So we want subclasses to never have to touch `__call__` at all, and trust that the top-level class is faithfully passing everything from `__call__` into `find_stars`. But if `__call__` is an abstract method, then subclasses *have* to implement their own `__call__`, so they might do that instead of `find_stars`.

Or put another way, either we shouldn't have `find_stars` *at all* (and instead tell users to just override `__call__`), or we should have `find_stars` be abstract but *fully implement* `__call__`.

Does that makes sense?